subsys/bluetooth/controller/hci_internal.c

```diff
@@ -310,6 +310,77 @@ static void supported_commands(sdc_hci_ip_supported_commands_t *cmds)
 #endif
 }
 
+#if defined(CONFIG_BT_HCI_VS)
+static void vs_supported_commands(sdc_hci_vs_zephyr_supported_commands_t *cmds)
+{
+	memset(cmds, 0, sizeof(*cmds));
+
+	cmds->read_version_info = 1;
+	cmds->read_supported_commands = 1;
+
+#if defined(CONFIG_BT_HCI_VS_EXT)
+	cmds->write_bd_addr = 1;
+	cmds->read_static_addresses = 1;
+	cmds->read_chip_temperature = 1;
+
+#if defined(CONFIG_BT_CTLR_TX_PWR_DYNAMIC_CONTROL)
+	cmds->write_tx_power_level = 1;
+#endif /* CONFIG_BT_CTLR_TX_PWR_DYNAMIC_CONTROL */
+#endif /* CONFIG_BT_HCI_VS_EXT */
+}
+#endif	/* CONFIG_BT_HCI_VS */
+
+static void supported_features(sdc_hci_ip_lmp_features_t *features)
+{
+	memset(features, 0, sizeof(*features));
+
+	features->bdedr_not_supported = 1;
+	features->le_supported = 1;
+}
+
+static void le_supported_features(sdc_hci_le_le_features_t *features)
+{
+	memset(features, 0, sizeof(*features));
+
+#ifdef CONFIG_BT_CTLR_LE_ENC
+	features->le_encryption = 1;
```

Comment:
Features that are non-configurable in the controller and that are valid from controller to controller, should not be wrapped with ifdefs. Otherwise we may end up in a situation where the peer device thinks encryption and similar is supported, but host support is disabled. See Core v5.2, Vol 6, Part B, Section 4.6.

The features which should not be wrapped by ifdefs:
* le_encryption
* extended_reject_indication
* channel_selection_algorithm_2